### PR TITLE
Added unit tests for staff APIs.

### PR DESCRIPTION
**Pull request by ShifaSZ**
Added unit tests for sample_metadata_export, saved_variants_page and upload_qc_pipeline_output APIs.

### CODE REVIEW

File: seqr/views/apis/staff_api_tests.py [hunk 1]
```diff
@@ -375,6 +375,45 @@
...
+    '0006_001-1A_2	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	WES	[]	Unknown	Unknown	[0.05456973953775274,0.05595057043008969,-0.008577987268070675,0.010774467092638367,-0.032619925661112215,0.010606945679118955]	nfe	0.0000e+00	0.0000e+00	0.0000e+00	0.0000e+00	0.0000e+00	0.0000e+00	1.0000e+00	0.0000e+00	5.4570e-02	5.5951e-02	-8.5780e-03	1.0774e-02	-3.2620e-02	1.0607e-02	true	false	false	true	true	false	["n_deletion","n_insertion","n_snp"]	7.4155e-01	15263762	5319819	0	15076982	120766	66014	186780	536	209877	19336	23581	143515	66362	0	2.1626e+00	1.8294e+00	8.1998e-01	5.2956e-01	15263762	1.5007e+07	15142996\n',
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:415]: Only include the relevant columns here

File: seqr/views/apis/staff_api_tests.py [hunk 2]
```diff
@@ -375,6 +375,45 @@
...
     'inheritance_description-1': 'de novo'}
 
+EXPECTED_SAMPLE_METADATA_ROW = {
+    "MME": "N",
+    "affected_status": "Affected",
+    "ancestry": "",
+    "ancestry_detail": "",
+    "data_type": "WES",
+    "date_data_generation": "2017-02-05",
+    "dbgap_submission": "No",
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:385]: You need to choose a sample row with airtable data and gene level data, ideally a gene thats NOT "Known gene for phenotype"

File: seqr/views/apis/staff_api_tests.py [hunk 3]
```diff
@@ -531,3 +570,39 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+    def test_upload_qc_pipeline_output(self, mock_file_iter):
+        url = reverse(upload_qc_pipeline_output,)
+        _check_login(self, url)
+
+        mock_file_iter.return_value = SAMPLE_QC_DATA
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:608]: You need to test the content of all of those keys, as well as test that the expected Individual models got updated

File: seqr/views/apis/staff_api_tests.py (hunk 1 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py:415]: Sorry, I didn't understand the 'qc' data. Most of the un-relevant columns are removed.

File: seqr/views/apis/staff_api_tests.py (hunk 3 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py:608]: Okay

File: seqr/views/apis/staff_api_tests.py (hunk 2 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py:385]: I'm sure about "Known gene for phenotype" gene.

File: seqr/views/apis/staff_api_tests.py [hunk 4]
```diff
@@ -375,6 +376,80 @@
...
+    "MME": "Y",
+    "subject_id": "NA19675_1"
+  }
+
+SAMPLE_QC_DATA = [
+    'seqr_id	data_type	filter_flags	qc_platform	qc_pop	qc_metrics_filters\n',
+    '03133B_2	WES	[]	Standard Germline Exome v5	nfe	[]\n',
+    'NA19675	WES	[]	Unknown	nfe	["n_deletion","n_insertion","n_snp"]\n',
+    'NA19678	WES	[]	Unknown	amr	["n_deletion","n_insertion","n_snp"]\n',
+    'HG00732	WES	["coverage"]	Standard Exome Sequencing v4	nfe	["r_insertion_deletion"]\n',
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:433]: You should also keep in the `HS_PCT_TARGET_BASES_20X` column and the `AL_PCT_CHIMERAS`, make sure you have at least one row where there is a `coverage` filter in filter_flags (which you have) and at least one row with `chimera` (which you don't). Then you should be testing the filter_flags dict on the Individual has the value from those columns

File: seqr/views/apis/staff_api_tests.py [hunk 5]
```diff
@@ -531,3 +606,87 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
+        self.assertListEqual(response_json['info'], [
+            u'Parsed 4 exome samples',
+            u'Found and updated matching seqr individuals for 3 samples'
+        ])
+        self.assertListEqual(response_json['warnings'], [
+            u'The following 1 samples were skipped: 03133B_2',
+            u'The following filter flags have no known corresponding value and were not saved: coverage_exome',
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:675]: This is bad - there shold be coverage_exome. As I said above, you need to add back in the `HS_PCT_TARGET_BASES_20X` column to the mock data. You should also add in an actually fake flag into the the `filter_flag` column in the raw data - something like `not_real_flag`, and check that it is correctly reported here

File: seqr/views/apis/staff_api_tests.py [hunk 6]
```diff
@@ -531,3 +606,87 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+            u'The following population platform filters have no known corresponding value and were not saved: n_deletion, n_insertion, n_snp, r_insertion_deletion'
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:676]: This is similarly bad - you need to add back in the `sample_qc.*` columns to the mock raw data (`sample_qc. n_deletion`, `sample_qc. n_insertion`, etc) and then add an intentionally fake value into the  qc_metrics_filters column to check here

File: seqr/views/apis/staff_api_tests.py [hunk 7]
```diff
@@ -531,3 +606,87 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+                'The following filter flags have no known corresponding value and were not saved: coverage_exome'),
+             mock.call(
+                'The following population platform filters have no known corresponding value and were not saved: n_deletion, n_insertion, n_snp, r_insertion_deletion'),
+             mock.call('Found and updated matching seqr individuals for 3 samples'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        population = [ indiv.population for indiv in Individual.objects.filter(id__in=[1,5])]
+        population.append(Individual.objects.get(id = 2).population)
+        self.assertListEqual(population, ['NFE', 'NFE', 'AMR'])
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:692]: Please check for each indiviodual that they have the expected population - this is really hard to read. There are 3 edited individuals as a result of this call - for each one fetch it from the db and confirm it has the expected updates. This should also include checking filter_flags and filter_flags

File: seqr/views/apis/staff_api_tests.py (hunk 4 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py]: Updated

File: seqr/views/apis/staff_api_tests.py (hunk 5 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py]: Updated

File: seqr/views/apis/staff_api_tests.py (hunk 6 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py]: Updated

File: seqr/views/apis/staff_api_tests.py (hunk 7 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py]: Changed to test per individual.

File: seqr/views/apis/staff_api_tests.py [hunk 8]
```diff
@@ -531,3 +608,82 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
+        self.assertListEqual(response_json['info'], [
+            u'Parsed 6 exome samples',
+            u'Found and updated matching seqr individuals for 4 samples'
+        ])
+        self.assertListEqual(response_json['warnings'], [
+            u'The following 2 samples were skipped: MANZ_1169_DNA, NA',
+            u'The following filter flags have no known corresponding value and were not saved: not_real_flag, contamination',
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:675]: contamination is a real flag, it corresponds to the PCT_CONTAMINATION column

File: seqr/views/apis/staff_api_tests.py [hunk 9]
```diff
@@ -531,3 +608,82 @@ def test_anvil_export(self, mock_export_multiple_files):
...
+            u'The following population platform filters have no known corresponding value and were not saved: not_real_filter'
+        ])
+
+        indiv = Individual.objects.get(id = 1)
+        self.assertListEqual([indiv.filter_flags, indiv.pop_platform_filters, indiv.population], [None, {u'n_deletion': '10898', u'n_snp': '127706', u'r_insertion_deletion': '1.2572E+00', u'r_ti_tv': '1.8292E+00', u'n_insertion': '13701'}, 'AMR'])
```
Comment (hanars) [inline seqr/views/apis/staff_api_tests.py:680]: Do these as 3 separate assertions, one for each field. Same for all the individuals below

File: seqr/views/apis/staff_api_tests.py (hunk 8 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py:675]: Ok, I will add the contamination flag back.

File: seqr/views/apis/staff_api_tests.py (hunk 9 above)
Comment (ShifaSZ) [reply seqr/views/apis/staff_api_tests.py:680]: Okay

**Pull request merged**